upup/pkg/fi/cloudup/awsup/aws_cloud.go

```diff
@@ -491,33 +491,44 @@ func FindAutoscalingGroups(c AWSCloud, tags map[string]string) ([]*autoscaling.G
 	}
 
 	if len(asgNames) != 0 {
-		request := &autoscaling.DescribeAutoScalingGroupsInput{
-			AutoScalingGroupNames: asgNames,
-		}
-		err := c.Autoscaling().DescribeAutoScalingGroupsPages(request, func(p *autoscaling.DescribeAutoScalingGroupsOutput, lastPage bool) bool {
-			for _, asg := range p.AutoScalingGroups {
-				if !matchesAsgTags(tags, asg.Tags) {
-					// We used an inexact filter above
-					continue
-				}
-				// Check for "Delete in progress" (the only use of .Status)
-				if asg.Status != nil {
-					glog.Warningf("Skipping ASG %v (which matches tags): %v", *asg.AutoScalingGroupARN, *asg.Status)
-					continue
+		for i := 0; i < len(asgNames); i += 50 {
```

Comment:
nit: might be good to pull out the `50` as a variable or a constant; it would make it easier to change in future and also let us put a comment like 'DescribeAutoScalingGroups is limited to 50 names at a time`.  But just a nit :-)

I also thought there was a function to do this batching in the go stdlib, but I can't find it at the moment.